<?xml version="1.0"?>
<launch>
  <!-- rename namespace â†’ node_namespace -->
  <arg name="namespace"        default=""/>
  <arg name="marker_topic"     default="/markers_in"/>
  <arg name="pointcloud_topic" default="/pointcloud_out"/>
  <arg name="target_frame"     default="lidar_link"/>
  <arg name="publish_rate"     default="50.0"/>
  <arg name="xy_scale"         default="1"/>
  <arg name="z_scale"          default="2"/>

  <!-- group under the passed-in namespace arg -->
  <group ns="$(arg namespace)">
    <node name="marker_to_pointcloud_node"
          pkg="dynamic_obstacle_prediction"
          type="marker_to_pointcloud_node"
          output="screen">
      <remap from="/markers_in"      to="$(arg marker_topic)"/>
      <remap from="/pointcloud_out"  to="$(arg pointcloud_topic)"/>
      <param name="target_frame"  value="$(arg target_frame)"/>
      <param name="publish_rate"  value="$(arg publish_rate)"/>
      <param name="xy_scale"      value="$(arg xy_scale)"/>
      <param name="z_scale"       value="$(arg z_scale)"/>
    </node>
  </group>
</launch>
